server.rb

```diff
@@ -1,163 +1,185 @@
+#!/usr/bin/env ruby
 require 'sinatra'
-require 'haml'
 require 'json'
-require 'open-uri'
-require 'thread'
-#set :bind, '192.168.42.110'
-#set :server, :thin
-#enable :lock
-
-HEADERS_HASH = {"User-Agent" => "Ruby/#{RUBY_VERSION}"}
-
-@@data_lock = Mutex.new
-@@data = {}
-@@last_img_url = nil
-@@max_history = 10
-
-@@data[:ocado] = {
-	routes: {},
-	info: {},
-	history: {}
+
+MAX_HISTORY = 10
+DATA_LOCK = Mutex.new
```

Comment:
@@ variables have been replaced with constants (in some places they were never destructively assigned to anyway).